File: ssm-chaincode/ssm-chaincode-client/src/main/java/ssm/chaincode/client/repository/CoopRepository.java
package ssm.chaincode.client.repository;

import java.util.List;
import java.util.concurrent.CompletableFuture;

import okhttp3.ResponseBody;
import retrofit2.http.*;

public interface CoopRepository {

    @GET("/")
    CompletableFuture<ResponseBody> command(
            @Query("cmd") String cmd,
            @Query("channelid") String channelid,
            @Query("chaincodeid") String chaincodeid,
            @Query("fcn") String fcn,
            @Query("args") List<String> args
    );


    @GET("/blocks/{blockId}")
    CompletableFuture<ResponseBody> getBlock(
            @Path("blockId") Long blockId,
            @Query("channelid") String channelid
    );

    @GET("/transactions/{txId}")
    CompletableFuture<ResponseBody> getTransaction(
            @Path("txId") String txId,
            @Query("channelid") String channelid
    );

    @POST("/")
    CompletableFuture<ResponseBody> invoke(
          @Body CommandArgs invokeArgs
    );

}

File: ssm-chaincode/ssm-chaincode-client/src/main/kotlin/ssm/chaincode/client/repository/CoopRepository.kt
package ssm.chaincode.client.repository

import okhttp3.ResponseBody
import retrofit2.http.*
import java.util.concurrent.CompletableFuture

interface CoopRepository {
    @GET("/")
    fun command(
        @Query("cmd") cmd: String?,
        @Query("channelid") channelid: String?,
        @Query("chaincodeid") chaincodeid: String?,
        @Query("fcn") fcn: String?,
        @Query("args") args: List<String?>?,
    ): CompletableFuture<ResponseBody>

    @GET("/blocks/{blockId}")
    fun getBlock(
        @Path("blockId") blockId: Long?,
        @Query("channelid") channelid: String?,
    ): CompletableFuture<ResponseBody>

    @GET("/transactions/{txId}")
    fun getTransaction(
        @Path("txId") txId: String?,
        @Query("channelid") channelid: String?,
    ): CompletableFuture<ResponseBody>

    @POST("/")
    operator fun invoke(
        @Body invokeArgs: CommandArgs?,
    ): CompletableFuture<ResponseBody>
}